src/jit/core_templates.expr

```diff
@@ -787,31 +984,227 @@
     (^getf $caller MVMFrame code_ref)
     (const 0 ptr_sz))))
 
+(template: coerce_sI
+  (call (^func &MVM_coerce_sI)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)
+      (carg $2 ptr)) ptr_sz))
+
 (template: say
   (callv (^func &MVM_string_say)
     (arglist
       (carg (tc) ptr)
       (carg $0 ptr))))
 
+(template: getenvhash
+  (call (^func &MVM_proc_getenvhash)
+    (arglist
+      (carg (tc) ptr)) ptr_sz))
+
+(template: backendconfig
+  (call (^func &MVM_backend_config)
+    (arglist
+      (carg (tc) ptr)) ptr_sz))
+
 (template: assertparamcheck!
   (when (zr $0)
     (callv (^func &MVM_args_bind_failed)
       (arglist
         (carg (tc) ptr)))))
 
+(template: getuniname
+  (call (^func &MVM_unicode_get_name)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)) ptr_sz))
+
+(template: getuniprop_int
+  (call (^func &MVM_unicode_codepoint_get_property_int)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 int)) int_sz))
+
+(template: getuniprop_bool
+  (call (^func &MVM_unicode_codepoint_get_property_bool)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 int)) int_sz))
+
+(template: getuniprop_str
+  (call (^func &MVM_unicode_codepoint_get_property_str)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 int)) ptr_sz))
+
+(template: matchuniprop
+  (call (^func &MVM_unicode_codepoint_has_property_value)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 int)
+      (carg $3 int)) ptr_sz))
+
+(template: nativecallbuild
+  (call (^func &MVM_nativecall_build)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)
+      (carg $2 ptr)
+      (carg $3 ptr)
+      (carg $4 ptr)
+      (carg $5 ptr)
+      (carg $6 ptr)) int_sz))
+
+(template: nativecallinvoke
```

Comment:
Is it still fine to have this in here?

Reply:
nativecallinvoke will be used in cases when the native call itself is not JIT compiled. This will happen on the very first call to a native function (unless it's precompiled) or if there are arguments that are not yet handled by the JIT, like callbacks.
In the cases where the call is really fast, nativeinvoke_o and friends are used.

Reply:
It never hurts :smile: